app/main.py

```diff
@@ -1 +1,45 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+    alive: list[Animal] = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False
+                 ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        self.add_to_alive(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+    @classmethod
+    def add_to_alive(cls, animal: Animal) -> None:
+        cls.alive.append(animal)
+
+    @classmethod
+    def remove_from_alive(cls, animal: Animal) -> None:
+        cls.alive.remove(animal)
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if self.hidden:
+            self.hidden = False
+        else:
+            self.hidden = True
```

Comment:
The `hide` method logic is inverted. The method should set `self.hidden` to `True` if it is currently `False` and vice versa. Consider using `self.hidden = not self.hidden` for a more concise implementation.